packages/remix/src/infrastructure/logto/handle-sign-in.ts

```diff
@@ -0,0 +1,34 @@
+import { Session } from '@remix-run/node';
+
+import { CreateLogtoClient } from './create-client';
+import { LogtoStorage } from './create-storage';
+
+type HandleSignInRequest = {
+  readonly redirectUri: string;
+};
+
+type HandleSignInResponse = {
+  readonly session: Session;
+  readonly navigateToUrl: string;
+};
+
+export const makeHandleSignIn =
+  (deps: { storage: LogtoStorage; createClient: CreateLogtoClient }) =>
+  async (request: HandleSignInRequest): Promise<HandleSignInResponse> => {
+    const { storage, createClient } = deps;
+
+    // eslint-disable-next-line @silverhand/fp/no-let
+    let navigateToUrl = '/';
+
+    const client = createClient((url) => {
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      navigateToUrl = url;
+    });
```

Comment:
Hmmm, I think I get it now. But can we make `makeLogtoClient` a class, and store the navigateUrl in the class as a private property?
In this way, we don't need to create the variable here with `let` and also don't have to pass an empty `navigate` function when initiatiate the `LogtoClient` in [here](https://github.com/logto-io/js/pull/417/files#r996918007)?

Reply:
You can refer to our [Next.js SDK](https://github.com/logto-io/js/blob/master/packages/next/src/index.ts#L15) and let me know if that makes sense to you.